homeassistant/components/tuya/sensor.py

```diff
@@ -734,6 +736,55 @@ class TuyaSensorEntityDescription(SensorEntityDescription):
             native_unit_of_measurement=UnitOfElectricPotential.VOLT,
             subkey="voltage",
         ),
+        TuyaSensorEntityDescription(
+            key=DPCode.VO,
+            name="Voltage",
+            device_class=SensorDeviceClass.VOLTAGE,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+            subkey="voltage",
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.CTIME,
+            name="Total charging time",
+            icon="mdi:history",
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.CO,
+            name="Temperature",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.KW,
+            name="Power",
+            device_class=SensorDeviceClass.POWER,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+            subkey="power",
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.CPOWER,  # Even if the device calls it power, this is the energy (kwh) consumed
```

Comment:
How does this work? As in, it seems non-standard? Tuya generally doesn't provide this data through normal DPs, so is this manufacturer/device specific?

Reply:
The device provide actual power used in KW, and the total amount od energy consumed since the charge at begun in KWH.
Screenshot:

![image](https://github.com/home-assistant/core/assets/19590516/32627b06-c2b5-40c8-9ffb-76ddc34224b6)

Reply:
Right, but that seems device/manufacturer specific?

We don't want to add device/manufacturer specific things in our integration, we should stick with the standards.